### ISSUE TITLE
Add validation dataset for LGBM models

### ISSUE THREAD

**Issue by LexABzH** [OSS-Pole-Emploi/gabarit#79]
### Problem

LGBM library allows to use a validation set in the fit function, but we use it inside a sklearn pipeline that does not implement validation sets.

## Concerned template

  - [x] NLP template
  - [x] NUM template
  - [ ] VISION template
  - [ ] How templates are generated - Jinja

### Solution

Rework the LGBM models so that we can use a validation dataset in the fit function.

- **Preferred solution:** Override the fit function for the LGBM models

- **Alternatives solution:** Do not inherit `ModelPipeline` but `ModelClass`
